### ISSUE TITLE
Removing dependency on interp1d?

### ISSUE THREAD

**Issue by EiffL** [pmelchior/spender#54]
In another project I ran into difficulties with `interp1d` which crashed torch for some reason when trying to get a gradient through it, so I just rewrote the interpolation routine myself like so:

```python
import torch

def interp1d(
    x, y, xnew,
    mask_value: float = 0.0):
    """Linear interpolation of a 1-D tensor using torch.searchsorted.
    Assumes that x and xnew are sorted in increasing order.

    Args:
        x (torch.Tensor): The x-coordinates of the data points, shape [batch, N].
        y (torch.Tensor): The y-coordinates of the data points, shape [batch, N].
        xnew (torch.Tensor): The x-coordinates of the interpolated points, shape [batch, M].
        mask_value (float): The value to use for xnew outside the range of x.
    Returns:
        torch.Tensor: The y-coordinates of the interpolated points, shape [batch, M].
    """
    indices = torch.searchsorted(x, xnew) - 1

    indices = torch.clamp(indices, 0, x.shape[1] - 1 - 1)

    slopes = (y[:, :-1] - y[:, 1:]) / (x[:, :-1] - x[:, 1:])

    # Interpolate the y-coordinates
    ynew = torch.gather(y, 1, indices) + (
        xnew - torch.gather(x, 1, indices)
    ) * torch.gather(slopes, 1, indices)

    # Mask out the values that are outside the valid range
    mask = (xnew < x[..., 0].reshape(-1, 1)) | (xnew > x[..., -1].reshape(-1, 1))
    ynew[mask] = mask_value

    return ynew
```
And this works completely fine.

Would you be open to trying to use this instead in spender? This way we could pip install it easily (in particular as a dependency for AstroPile benchmarks....)

**Comment by EiffL** [comment]
Well.... this apparently happened just 17 hours ago, maybe no need to change anything:
https://pypi.org/project/torchinterp1d/

**Comment by pmelchior** [comment]
I'm aware: https://github.com/aliutkus/torchinterp1d/issues/18

**Comment by EiffL** [comment]
ahahah, perfect :-) All good then
